File: adt2unit/pas/init_old_songdata.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure init_old_songdata;

{$IFNDEF ADT2PLAY}
var
  temp: Byte;
{$ENDIF} // NOT DEFINED(ADT2PLAY)

begin
  _dbg_enter ({$I %FILE%}, 'init_old_songdata');

  FillChar(old_songdata,SizeOf(old_songdata),0);
  FillChar(old_songdata.pattern_order,SizeOf(old_songdata.pattern_order),$080);
  FillChar(old_songdata.instr_data,SizeOf(old_songdata.instr_data),0);

{$IFNDEF ADT2PLAY}
  For temp := 1 to 250 do
    old_songdata.instr_names[temp] :=
      ' iNS_'+byte2hex(temp)+#247' ';
{$ENDIF} // NOT DEFINED(ADT2PLAY)

  _dbg_leave; //EXIT //init_old_songdata
end;

File: adt2unit/init_old_songdata.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2026 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void init_old_songdata (void)
{
  tOLD_FIXED_SONGDATA *song = &old_songdata;

  DBG_ENTER ("init_old_songdata");

  SetLength (song->songname, 0);
  SetLength (song->composer, 0);
#if !ADT2PLAY
  for (uint8_t i = 0; i < /*INSTRUMENTS_MAX_A2M_V1*/250; i++)
  {
    SetLength (song->instr_names[i], 9);
    snprintf (GetStr ((char *)song->instr_names[i]), 1 + 9, " iNS_%02"PRIX8"%c ", i + 1, /*charmap.*/'\xF7');
    memset (&song->instr_data[i], 0, sizeof (song->instr_data[0]));
  }
#endif // !ADT2PLAY
  memset (song->pattern_order, PATTERN_ORDER_JUMP + 0, sizeof (song->pattern_order));
  song->tempo = 0;
  song->speed = 0;
  song->common_flag = 0;

  DBG_LEAVE (); //EXIT //init_old_songdata
}
